File: ai-sentinel-core/src/main/java/dev/aisentinel/core/contract/ContractSecurityAdvice.java
package dev.aisentinel.core.contract;

import java.util.List;
import java.util.Objects;

/**
 * Transport-safe advisory. Explicitly non-authoritative for enforcement.
 */
public record ContractSecurityAdvice(
    String code,
    String priority,
    String reason,
    List<String> linkedFactorCodes,
    boolean humanReviewRecommended
) {
    public ContractSecurityAdvice {
        Objects.requireNonNull(code, "code");
        Objects.requireNonNull(priority, "priority");
        reason = reason == null ? "" : reason;
        linkedFactorCodes = linkedFactorCodes == null ? List.of() : List.copyOf(linkedFactorCodes);
    }
}

File: dotnet/src/AI.Sentinel.AspNetCore/Contract/ContractSecurityAdvice.cs
using System.Text.Json.Serialization;

namespace AI.Sentinel.AspNetCore.Contract;

public sealed class ContractSecurityAdvice
{
    [JsonPropertyName("code")]
    public string Code { get; set; } = string.Empty;

    [JsonPropertyName("priority")]
    public string Priority { get; set; } = string.Empty;

    [JsonPropertyName("reason")]
    public string Reason { get; set; } = string.Empty;

    [JsonPropertyName("linkedFactorCodes")]
    public List<string> LinkedFactorCodes { get; set; } = new();

    [JsonPropertyName("humanReviewRecommended")]
    public bool HumanReviewRecommended { get; set; }
}
